src/src/Helpers/HTTPResponseParser.cpp

```diff
@@ -0,0 +1,397 @@
+        lastJsonMessageLength = message.length() * 2;
+      }
+
+      // Allocate memory for root if needed
+      if (root == nullptr) {
+  #  ifdef USE_SECOND_HEAP
+        HeapSelectIram ephemeral;
+  #  endif // ifdef USE_SECOND_HEAP
+        root = new (std::nothrow) DynamicJsonDocument(lastJsonMessageLength);
+      }
+
+      if (root != nullptr) {
+        // Parse the JSON
+        DeserializationError error = deserializeJson(*root, message);
+
+        if (!error) {
+          // Process the keys from the file
+          readAndProcessJsonKeys(root, numJson);
+        } else {
+          addLog(LOG_LEVEL_ERROR, strformat(F("Parsing JSON failed: %s"), error.c_str()));
+        }
+
+        // Cleanup JSON resources
+        cleanupJSON();
+      }
+    }
+    # endif // if FEATURE_JSON_EVENT
+  }
+}
+
+// ------------------------------------------------------------------------------------------- JSONevent Key processing
+   # if FEATURE_JSON_EVENT
+
+void readAndProcessJsonKeys(DynamicJsonDocument *root, int numJson) {
+#  if FEATURE_USE_DOUBLE_AS_ESPEASY_RULES_FLOAT_TYPE
+  int nr_decimals = ESPEASY_DOUBLE_NR_DECIMALS;
+#  else // ifdef FEATURE_USE_DOUBLE_AS_ESPEASY_RULES_FLOAT_TYPE
+  int nr_decimals = ESPEASY_FLOAT_NR_DECIMALS;
+#  endif // ifdef FEATURE_USE_DOUBLE_AS_ESPEASY_RULES_FLOAT_TYPE
+
+  // Open the `json.keys` file
+#  ifdef ESP8266
+  const String fileName = F("json.keys");
+#  else // ifdef ESP8266
+  const String fileName = F("/json.keys");
+#  endif // ifdef ESP8266
+
+  if (!fileExists(fileName)) {
+    addLogMove(LOG_LEVEL_ERROR, strformat(F("%s does not exist!"), fileName.c_str()));
+    return;
+  }
+  File keyFile = ESPEASY_FS.open(fileName, "r");
+
+  int keyCount                   = 0;
+  int successfullyProcessedCount = 0; // Counter for successfully processed keys
+  String csvOutput;                   // Collect values as a CSV string
+
+  while (keyFile.available()) {
+    String key = keyFile.readStringUntil('\n');
+    key.trim(); // Remove extra spaces or newlines
+
+    if (key.isEmpty()) {
+      continue;
+    }
+    int keyNumber = 0;
+
+    // Extract the number prefix (before the first dot)
+    if (key.indexOf(':') != -1) {
+      keyNumber = key.substring(0, key.indexOf(':')).toInt();
+      key       = key.substring(key.indexOf(':') + 1);
+    }
+
+    // Only process keys that match the filterNumber (or process all if no match)
+    if ((numJson > 0) && (keyNumber != numJson)) {
+      continue; // Skip this key if it doesn't match the filter number
+    }
+    keyCount++;
+
+    if (keyCount > MAX_KEYS) {
+      addLogMove(LOG_LEVEL_ERROR, strformat(F("Warning: More than %d keys in %s"), MAX_KEYS, fileName.c_str()));
+    }
+
+    // Process the key and navigate the JSON
+    JsonVariant value = *root;
+    size_t start = 0, end;
+
+    while ((end = key.indexOf('.', start)) != (unsigned int)-1) {
+      String part = key.substring(start, end);
+      value = value[part];
```

Comment:
`String` variables that won't change can be declared `const String`, so the compiler can often optimize them better.